src/openforms/contrib/haal_centraal/tests/test_integration.py

```diff
@@ -0,0 +1,147 @@
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+from django.test import TestCase
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.authentication.utils import store_auth_details, store_registrator_details
+from openforms.config.models import GlobalConfiguration
+from openforms.forms.tests.factories import FormStepFactory
+from openforms.prefill import prefill_variables
+from openforms.submissions.tests.factories import SubmissionStepFactory
+from openforms.typing import JSONValue
+from openforms.utils.tests.vcr import OFVCRMixin
+
+from ..models import HaalCentraalRequestOptions
+from .factories import HaalCentraalConfigFactory
+
+CONFIGURATION = {
+    "display": "form",
+    "components": [
+        {
+            "key": "bsn",
+            "type": "textfield",
+            "label": "BSN",
+            "prefill": {
+                "plugin": "haalcentraal",
+                "attribute": "burgerservicenummer",
+            },
+            "multiple": False,
+        },
+        {
+            "key": "address",
+            "type": "textfield",
+            "label": "Address",
+            "prefill": {
+                "plugin": "haalcentraal",
+                "attribute": "adressering",
+            },
+            "multiple": False,
+        },
+    ],
+}
+
+
+class BRPIntegrationTest(OFVCRMixin, TestCase):
+    """An integration test against a test instance, to make sure headers are set correctly.
+    The prefill feature is being used here, although any other feature making use of the Haal Centraal API
+    could have been used.
+    """
+
+    VCR_TEST_FILES = Path(__file__).parent / "files"
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        global_config = GlobalConfiguration(organization_oin="00000003273229750000")
+
+        global_config_patcher = patch(
+            "openforms.contrib.haal_centraal.clients.GlobalConfiguration.get_solo",
+            return_value=global_config,
+        )
+        global_config_patcher.start()
+        self.addCleanup(global_config_patcher.stop)
+
+        form_step = FormStepFactory.create(form_definition__configuration=CONFIGURATION)
+        self.submission_step = SubmissionStepFactory.create(
+            submission__form=form_step.form, form_step=form_step
+        )
```

Comment:
please look for examples of `SubmissionFactory.create(form__generate_minimal_example=True, form__formstep__form_definition__configuration=...)`

These ensure that all data is correctly wired up and let you reason about the submission directly instead of taking a detour via `submission_step`.